lib/services/elasticsearch.js

```diff
@@ -1286,26 +1327,29 @@ class ElasticSearch extends Service {
         extractedDocuments
       } = this._extractMDocuments(documents, kuzzleMeta);
 
-    for (let i = 0; i < extractedDocuments.length; i++) {
-      if (typeof extractedDocuments[i]._id === 'string') {
+    while ((extractedDocument = extractedDocuments.shift()) !== undefined) {
```

Comment:
![BLOCKER](https://sonarsource.github.io/sonar-github/severity-blocker.png 'Severity: BLOCKER') Add the "let", "const" or "var" keyword to this declaration of "extractedDocument" to make it explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.kaliop.net/coding_rules#rule_key=javascript%3AS2703)